slsDetectorSoftware/src/Detector.cpp

```diff
@@ -141,6 +141,55 @@ void Detector::setDetectorSize(const defs::xy value) {
     pimpl->setNumberOfChannels(value);
 }
 
+std::vector<defs::detectorSettings> Detector::getSettingsList() const {
```

Comment:
Sine each settings vector is unique for the detector we don't need to build it up using push_back. The vector can be constructed with the values needed:
```
case defs::EIGER:
     return std::vector<defs::detectorSettings>{defs::STANDARD, defs::HIGHGAIN ....}
```

Reply:
done